<launch>

  <!-- Launch the world -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find com3528_2025_team7)/worlds/football_pitch.world"/>
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
  
  </include>


  <!-- NEW CODE -->
  <node pkg="com3528_2025_team7" type="state_aggregator.py" name="state_aggregator" output="screen"/>

  <!-- NEW CODE -->
  <node pkg="com3528_2025_team7" type="velocity_aggregator.py" name="velocity_aggregator" output="screen" />

  <!-- 2) Agent 1 in namespace /miro01 -->
  <group ns="miro01">
    <env   name="MIRO_ROBOT_NAME" value="miro01"/>
    <param name="agent_id"         value="agent_1"/>
    <node  pkg="com3528_2025_team7"
           type="boid_main_v2.py"
           name="agent_1"
           output="screen"/>
  </group>

  <!-- 3) Agent 2 in namespace /miro02 -->
  <group ns="miro02">
    <env   name="MIRO_ROBOT_NAME" value="miro02"/>
    <param name="agent_id"         value="agent_2"/>
    <node  pkg="com3528_2025_team7"
           type="boid_main_v2.py"
           name="agent_2"
           output="screen"/>
  </group>

  <!-- 4) Agent 3 in namespace /miro03 -->
  <group ns="miro03">
    <env   name="MIRO_ROBOT_NAME" value="miro03"/>
    <param name="agent_id"         value="agent_3"/>
    <node  pkg="com3528_2025_team7"
           type="boid_main_v2.py"
           name="agent_3"
           output="screen"/>
  </group>


  <!-- 4) Agent 4 in namespace /miro03 -->
  <group ns="miro04">
    <env   name="MIRO_ROBOT_NAME" value="miro04"/>
    <param name="agent_id"         value="agent_4"/>
    <node  pkg="com3528_2025_team7"
           type="boid_main_v2.py"
           name="agent_4"
           output="screen"/>
  </group>

  
  <!-- 4) Agent 5 in namespace /miro03 -->
  <group ns="miro05">
    <env   name="MIRO_ROBOT_NAME" value="miro05"/>
    <param name="agent_id"         value="agent_5"/>
    <node  pkg="com3528_2025_team7"
           type="boid_main_v2.py"
           name="agent_5"
           output="screen"/>
  </group>

</launch>
